test/api/fake_api.dart

```diff
@@ -68,9 +68,9 @@ class FakeHttpClient extends http.BaseClient {
 
   @override
   Future<http.StreamedResponse> send(http.BaseRequest request) {
+    lastRequest = request;
     final response = _nextResponse!;
     _nextResponse = null;
-    lastRequest = request;
```

Comment:
```
fake_api: Record premature requests, too, in lastRequest
```

Ah I think I see: a request is "premature" because it's made before its response has been prepared. So the difference is that in those cases (the cases where `_nextResponse!` fails), the premature request will be recorded as `lastRequest`, vs. not recorded.

Reply:
Yep, exactly.